### ISSUE TITLE
nwr and recurse 

### ISSUE THREAD

**Issue by mmd-osm** [drolbr/Overpass-API#504]
I tried to convert the example https://wiki.openstreetmap.org/wiki/Overpass_API/Overpass_API_by_Example#Find_house_numbers_without_street to nwr style. 

```c++
area[type=boundary]["de:amtlicher_gemeindeschluessel"="08115003"]->.boundaryarea;
rel(area.boundaryarea)[type=associatedStreet]->.associatedStreet;

nwr(area.boundaryarea)["addr:housenumber"][!"addr:street"][!"addr:place"]->.allHouses;
nwr(r.associatedStreet:"house")->.asHouse;
((.allHouses; - .asHouse;); >; );out meta;
```

Unfortunately, this didn't work out and returned the following error message. I think error message is not accurate, i.e. it should be possible to return nodes, ways and relations with a certain role in one step this way.

```
Error: line 5: parse error: A recursion from type 'r' produces nodes, ways, or relations.
Error: line 5: static error: For the attribute "type" of the element "recurse" the only allowed values are "relation-relation", "relation-backwards","relation-way", "relation-node", "way-node", "way-relation","node-relation", "node-way", "down", "down-rel", "up", or "up-rel".
Error: line 5: static error: A role can only be specified for values "relation-relation", "relation-backwards","relation-way", "relation-node", "way-relation",or "node-relation". 
```

**Comment by drolbr** [comment]
Yes, that makes sense.

**Comment by drolbr** [comment]
Fixed in 842d938c73feec69867a7c6f5e64e1c7015d60d0
